Upstream-Status: Pending

GCC 5.x defaults to -std=gnu11 instead of -std=gnu89. The semantics for
inline functions changes which causes build failures such as:

| misc.h:68:20: warning: inline function 'swaps' declared but never defined
|  extern inline void swaps (void *, int);
|                      ^

and

| control.c:1717: undefined reference to `check_control'

Fix them by:
1 drop 'inline' attribute for function swap()
2 replace 'inline' with 'static inline' for other functions

Signed-off-by: Kai Kang <kai.kang@windriver.com>
---
 control.c | 8 ++++----
 misc.c    | 2 +-
 misc.h    | 2 +-
 network.c | 4 ++--
 4 files changed, 8 insertions(+), 8 deletions(-)

diff --git a/control.c b/control.c
index 134a7bf..0ef505b 100644
--- a/control.c
+++ b/control.c
@@ -1139,7 +1139,7 @@ int control_finish (struct tunnel *t, struct call *c)
     return 0;
 }
 
-inline int check_control (const struct buffer *buf, struct tunnel *t,
+static inline int check_control (const struct buffer *buf, struct tunnel *t,
                           struct call *c)
 {
     /*
@@ -1275,7 +1275,7 @@ inline int check_control (const struct buffer *buf, struct tunnel *t,
     return 0;
 }
 
-inline int check_payload (struct buffer *buf, struct tunnel *t,
+static inline int check_payload (struct buffer *buf, struct tunnel *t,
                           struct call *c)
 {
     /*
@@ -1381,7 +1381,7 @@ inline int check_payload (struct buffer *buf, struct tunnel *t,
 #endif
     return 0;
 }
-inline int expand_payload (struct buffer *buf, struct tunnel *t,
+static inline int expand_payload (struct buffer *buf, struct tunnel *t,
                            struct call *c)
 {
     /*
@@ -1561,7 +1561,7 @@ void send_zlb (void *data)
     toss (buf);
 }
 
-inline int write_packet (struct buffer *buf, struct tunnel *t, struct call *c,
+static inline int write_packet (struct buffer *buf, struct tunnel *t, struct call *c,
                          int convert)
 {
     /*
diff --git a/misc.c b/misc.c
index 3092401..af90dbf 100644
--- a/misc.c
+++ b/misc.c
@@ -170,7 +170,7 @@ void do_packet_dump (struct buffer *buf)
     printf ("}\n");
 }
 
-inline void swaps (void *buf_v, int len)
+void swaps (void *buf_v, int len)
 {
 #ifdef __alpha
     /* Reverse byte order alpha is little endian so lest save a step.
diff --git a/misc.h b/misc.h
index aafdc62..caab7a1 100644
--- a/misc.h
+++ b/misc.h
@@ -65,7 +65,7 @@ extern void l2tp_log (int level, const char *fmt, ...);
 extern struct buffer *new_buf (int);
 extern void udppush_handler (int);
 extern int addfcs (struct buffer *buf);
-extern inline void swaps (void *, int);
+extern void swaps (void *, int);
 extern void do_packet_dump (struct buffer *);
 extern void status (const char *fmt, ...);
 extern void status_handler (int signal);
diff --git a/network.c b/network.c
index 5e5b8b2..b019c2c 100644
--- a/network.c
+++ b/network.c
@@ -135,7 +135,7 @@ int init_network (void)
     return 0;
 }
 
-inline void extract (void *buf, int *tunnel, int *call)
+static inline void extract (void *buf, int *tunnel, int *call)
 {
     /*
      * Extract the tunnel and call #'s, and fix the order of the 
@@ -155,7 +155,7 @@ inline void extract (void *buf, int *tunnel, int *call)
     }
 }
 
-inline void fix_hdr (void *buf)
+static inline void fix_hdr (void *buf)
 {
     /*
      * Fix the byte order of the header
-- 
1.9.1

